DOCS-5309 Vault Integration Edits + Metrics Versioning Organization (#15069)
DataDog/integrations-core: vault/README.md

> #### Prerequisites

Author: ```suggestion
```

> This OpenMetrics-based integration has a latest version (`use_openmetrics`: true) and a legacy version (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. Versions 3.4.0 and later of this check use [OpenMetrics][21] for metric collection, which requires Python 3. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][22].

Reviewer: I feel like the word "version" gets a bit overloaded in this paragraph, and the `use OpenMetrics for metric collection` actually means using *latest*, which is unclear from that wording.

I've given it some thought and I think something like the following would add clarity:

```suggestion
Starting with version 3.4.0, this OpenMetrics-based integration has a *latest* mode (`use_openmetrics`: true) and a *legacy* mode (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends enabling the *latest* mode. Note that *latest* mode requires Python 3. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][22].
```

I'm still not sure about the mention to Python 3, though, since it looks like most of the other integrations don't mention it.

Let me know what you think.

#### Prerequisites

> This OpenMetrics-based integration has a latest version (`use_openmetrics`: true) and a legacy version (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][22].

Author: ```suggestion
This OpenMetrics-based integration has a latest version (`use_openmetrics`: true) and a legacy version (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. Versions 3.4.0 and later of this check use [OpenMetrics][21] for metric collection, which requires Python 3. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][22].
```

> This OpenMetrics-based integration has a latest version (`use_openmetrics`: true) and a legacy version (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. For more information, see the [Latest and Legacy Versioning For OpenMetrics-based Integrations][22].

Author: ```suggestion
This OpenMetrics-based integration has a latest version (`use_openmetrics`: true) and a legacy version (`use_openmetrics`: false). To get all the most up-to-date features, Datadog recommends upgrading to the latest version. For more information, see [Latest and Legacy Versioning For OpenMetrics-based Integrations][22].
```

1. Ensure you have enabled [Prometheus metrics in the Vault configuration][3].

>    * [Metrics starting with `vault.replication.fetchRemoteKeys`, `vault.replication.merkleDiff`, and `vault.replication.merkleSync`](#data-collected) are not reported unless the replication is in an unhealthy state. 

Reviewer: I'm unsure of the placement of this paragraph. It does look like it belongs close to where the collected metrics are listed.

Author: ```suggestion
```

>    * Versions 3.4.0 and later of this check use [OpenMetrics][21] for metric collection, which requires Python 3. For hosts that are unable to use Python 3, or if you would like to use a legacy version of this check, set the value of `use_openmetrics` to `false` in the configuration.

Reviewer: I think I would integrate this paragraph with the first one about latest vs legacy, as it's otherwise a bit redundant. We should also probably try to avoid using the bare OpenMetrics name to refer to the V2 behavior.

Author: ```suggestion
```

See [metadata.csv][14] for a list of metrics provided by this integration.

> ### Events

Author: ```suggestion

* [Metrics starting with `vault.replication.fetchRemoteKeys`, `vault.replication.merkleDiff`, and `vault.replication.merkleSync`](#data-collected) are not reported unless the replication is in an unhealthy state. 

### Events
```

[19]: https://www.datadoghq.com/blog/vault-monitoring-tools
[20]: https://www.datadoghq.com/blog/vault-monitoring-with-datadog
[21]: https://docs.datadoghq.com/integrations/openmetrics/

> [22]: https://docs.datadohgq.com/integrations/guide/versions-for-openmetrics-based-integrations

Reviewer 2: ```suggestion
[22]: https://docs.datadoghq.com/integrations/guide/versions-for-openmetrics-based-integrations
```
